### ISSUE TITLE
i18n for templates

### ISSUE THREAD

**Issue by vksiva** [fizzed/rocker#78]
i18n support - Is there a way that rocker compilation can be modified to generate java classes for  different locales based on message bundles?

**Comment by jjlauer** [comment]
hey @vksiva -- can you give a more concrete example of what you're trying to accomplish?  You can currently just create a static java method you import and do lookups at runtime.  For example, create a static public method like `i18n(String key)` and then in your template do stuff like `@i18n("my-key")`.  If you're thinking of something else, happy to help if you elaborate with more examples.

**Comment by vksiva** [comment]
Thanks for the tip @jjlauer!  
I guess I could do that, but is there a way I could create them (pages/java/class files) upfront for each of the message bundle during compile time. 
Since the text labels are not going to change ... I was just exploring.

**Comment by jjlauer** [comment]
So you want a rocker template to be compiled into another rocker template
per message bundle.  Something like "view.rocker.html" into various
"view_en.java" files that then get compiled.  You could probably try using
the "Post Processing" pipeline to put something together.  For example,
you'd create a "view_en.rocker.html" rocker template, then use
post-processing to look the message label up so its slipstreamed into the
.java generated file.

Unless your i18n messages are really large, I'm not sure you'd gain
substantial performance having them directly compiled into a Rocker
template.  That being said, letting the compiler verify a message exists
for a particular key would be pretty useful.  Maybe not a bad idea to look
at allowing Rocker to compile message bundles into Rocker variables that
can be used directly in templates.

**Comment by vksiva** [comment]
Yes, thats exactly what I wanted. Single Template -> 1 Java file per message bundle.
I will try to use the post processing hook to get the desired result. 

Never realized that compile time check of the message keys would be a very beneficial feature (At least from the testing point of view).  

Thanks again for the insights.
